### PR TITLE
Change icon

### PR DESCRIPTION
**Pull request by QuestYouCraft**
Changes:
- changed icon
- ~~deleted open browser & redirect to configuration.md actions (because the config file already says where to go to get help on the config file)~~

### CODE REVIEW

**Comment by HerMajestyDrMona** [comment]
Sorry, the program icon won't be changed in the main release. It is supposed to be this way intentionally.

Is the provided new icon file [copyrighted?](https://www.iconfinder.com/icons/1582904/drag_drag_and_drop_drag&drop_drop_move_replace_icon) Are you sure it would be allowed to store such an icon in the open source project, where .ico file can be viewed and downloaded by anyone?

I have permission from the author of the current icon to use it in this project, so I find it as a safer solution. Besides that it makes the program look cooler, way better than the other common icons. It brings more attention. So applying this change to the main release would make me have to manually change it back for myself. I like when the software shows "my vision", I would rather keep the cute Mona Lisa Megan Fox icon. It's way more recognizable and it harder to mistake with the other similar icons (like Vmware tray icon).

However, it might be a good idea to add an alternative icon to the program, for people who dislike the Mona Lisa Megan Fox. That could be done with a configuration file. This way everyone would be happy. So as soon as I get details about the provided icon licensing, I'll try to figure something out :)

I'm not sure about the "ShellExecuteW" deletion that opens GitHub page with the configuration guide. After all, copying the link to clipboard and then pasting it to the browser window is more time consuming and less intuitive. The link in the .txt file was provided in case when ShellExecute to open the configuration guide fails. In rare cases, users have misconfigured default Internet browser settings, and http / https links simply won't open. In this case they are still able to find the link at the bottom of the default created config file.
Another thing is, the `Windows11DragAndDropToTaskbarFixConfig.txt` could be created on an older version of the program. In this case, after clicking on the "Configure..." option, the link in browser would not open, but also the link to the configuration would be unavailable. Of course, it can be changed by opening the config file, searching for the missing GitHub link, writing at the end of the file. But then again, it could fail in cases where program was pasted to the protected folder, and ask for administrator permissions in order to be able to do it.

So in summary, the simplest way is to keep it as it is currently.

Thank you for the idea and contribution. I agree that there are a lot of things to weak when time allows it. I was very speeding up with the recent versions in order to provide it to users as soon as possible. But in this case, I think it should stay as it is. Of course please feel free to share some arguments for removing it, so it can be reconsidered.

**Comment by QuestYouCraft** [comment]
> Is the provided new icon file copyrighted?

The icon, although similar, is not the same as the one you provided in the hyperlink. Therefore, the icon, in theory, is protected by my copyright, but not someone else's. If you want, I can show the differences and prove that the copyright on the icon is mine.

> However, it might be a good idea to add an alternative icon to the program

Yes, it's a excellent idea and my icon is perfect for this idea (at least as an alternative icon).

> I'm not sure about the "ShellExecuteW" deletion that opens GitHub page with the configuration guide.

But I am sure of it, because not all PCs have access to the Internet. Moreover, it may be unexpected for the user that when opening the config file, the browser window also opens with it. I am not sure that the user will open the link immediately after reading the config file. Even if this is the case, in many text editors, you can select a link and click on it from the context menu. Therefore, in essence, ShellExecW is not needed.

**Comment by HerMajestyDrMona** [comment]
> The icon, although similar, is not the same as the one you provided in the hyperlink. Therefore, the icon, in theory, is protected by my copyright, but not someone else's. If you want, I can show the differences and prove that the copyright on the icon is mine.

I can see minor difference with lines sharpness and some cuts, but I wonder if it's enough? Maybe it's better to recreate something that does not look as similar to these paid ones? In my understanding, if you for example edit an audio track (adding sound effects) and upload on YouTube, you will still get copyright claim from the owner. You could then open a dispute, etc. However, I think it's simply better to avoid potential situations like this, than spend time later on explanations? I will ask a friend who creates awesome graphics if he can figure something out.

> But I am sure of it, because not all PCs have access to the Internet. Moreover, it may be unexpected for the user that when opening the config file, the browser window also opens with it. I am not sure that the user will open the link immediately after reading the config file. Even if this is the case, in many text editors, you can select a link and click on it from the context menu. Therefore, in essence, ShellExecW is not needed.

The config file created before version 1.8 does not contain the GitHub page link. Same with config files created manually by users. Therefore all config variables would need to be moved to the .txt file, together with description of what these options do. At this moment I did not have time to polish all texts, so it seemed like a better idea to open a web page which can be updated "live". When it comes to Internet access, all 3 previous options in menu also open links. So it's not like the `Configure...` option does something abnormal.
When it comes to the text editors, I noticed that it always uses Windows Notepad, because one of the ShellExecute parameters is "edit" instead of "open". So these links can't be clicked in any easy way. I'm thinking about changing this parameter to "open", so Notepad++ or others could open it, however then graying out the "Configure...." option might stop working, as different text editor would take the handle. One of solutions would be checking if Internet is active before opening these links, but that could cause other programs, trigger firewall messages, etc. It's better to simplicity things like this in a software of this type.
Another aspect is that configuring the program is not something frequently done. I think most of users will only open it 1 time, therefore we shouldn't worry that much about this option. We could add a new config option that disables Github page opening on clicking that button though. The question is if more configuration variables for minor things like this won't "disgust" users from reading the CONFIGURATION page, because it's so much text to read :)

**Comment by QuestYouCraft** [comment]
> I will ask a friend who creates awesome graphics if he can figure something out.

Looking forward to hearing back.

> I'm thinking about changing this parameter to "open"...

ShellExecuteW parameter has already been changed by you to "open".

> I think most of users will only open it 1 time, therefore we shouldn't worry that much about this option.

Ok, I'm agree with it. You've convinced me.

**Comment by HerMajestyDrMona** [comment]
> ShellExecuteW parameter has already been changed by you to "open".

I meant the Shell Execute function for opening `Windows11DragAndDropToTaskbarFixConfig.txt`:
https://github.com/HerMajestyDrMona/Windows11DragAndDropToTaskbarFix/blob/003fb2e6a6c27085b3a9de8cc8365b7e7b7729fa/Windows11DragAndDropToTaskbarFix/Windows11DragAndDropToTaskbarFix.cpp#L3552

The "edit" parameter makes it use the normal Notepad instead of Notepad++ (if set as default for .txt files). I mentioned it as an argument for:
> Even if this is the case, in many text editors, you can select a link and click on it from the context menu.

**Comment by QuestYouCraft** [comment]
Ok, but what about the logo?

**Comment by HerMajestyDrMona** [comment]
> Ok, but what about the logo?

What do you think about this one? https://www.veryicon.com/icons/system/gesture-series/drag-drop-1.html

"License: Free for personal and commercial purpose."

I would just mention where it comes from. I quite like the hand on the icon.

**Comment by QuestYouCraft** [comment]
I'm leaning towards the second icon:

> https://www.veryicon.com/icons/system/gesture-series/drag-drop.html

But it also needs to be modified (change the color). By 22 UTC I will edit the icon.

**Review by motcke — APPROVED** [review]

**Comment by HerMajestyDrMona** [comment]
Thank you. I will check it in the following days. I'm a bit overloaded currently and have migraine most of the time. I think these changes are anyway very minor that they can wait for a little bit. I will be trying to unlock the "drop to pin" thing as well before the next version release.

**Comment by QuestYouCraft** [comment]
> ...have migraine most of the time.

Get well.

**Comment by HerMajestyDrMona** [comment]
Hi again. I tested the icon and it seems to be super invisible:
Screenshot 2021-11-11 202801

We would need it to have a color that is noticeable with both: white and dark theme of the taskbar. Also the margins can't be as big as they are now, because the icon looks smaller than all others.

**Comment by QuestYouCraft** [comment]
Excuse me for the late response. I completely agree with you, I also thought about this when I created the icon. I will make a new adapted icon, compile and test based on the code of the latest release. At 22:00 UTC (13.11.2021) I will give an answer.

**Comment by QuestYouCraft** [comment]
I made 5 builds of the program with different colors. I stopped at this option (it differs from the old one in color).

**Comment by HerMajestyDrMona** [comment]
Thanks! It's not so perfect, but looks acceptable. As I mentioned earlier, this will be an alternative icon anyway.

Could you please create a pull request with only new file added: "Windows11DragAndDropToTaskbarFixAlternative.ico" (your new icon)? I have some other changes in the code for loading the alternative icon, etc. I don't want to replace README and CONFIGURATION stuff, as there only will be added information about the alternative icon option.

I didn't manage a GitHub project before, so I don't know if I could do these changes myself in the pull request.

**Comment by QuestYouCraft** [comment]
Is that okay? It is important for me that my name is mentioned.

**Comment by HerMajestyDrMona** [comment]
Generally, this pull request has many changes and reverts. I think it would be better to open a new one with the Alternative icon file only. I only asked, because you then would appear on the Contributors tab if I accepted it.
I need to update the config, readme, and such anyway for the next version, so it's not necessary to have them all updated in this pull request. The veryicon.com source is also missing from the README mention, so I will need to add it anyway. If possible, please just create a new pull request with "Windows11DragAndDropToTaskbarFixAlternative.ico", and I'll do the rest.

**Comment by QuestYouCraft** [comment]
Ok
